src/theory/strings/inference_manager.cpp

```diff
@@ -625,6 +626,16 @@ void InferenceManager::explain(TNode literal,
     }
   }
 }
+void InferenceManager::setIncomplete() { d_out.setIncomplete(); }
+
+void InferenceManager::markCongruent(Node a, Node b)
+{
+  ExtTheory* eth = d_parent.getExtTheory();
+  if (eth->hasFunctionKind(a.getKind()))
```

Comment:
Shouldn't we also check the kind of `b`?

Reply:
They should be the same, I'll add an assertion.